lib/Doctrine/Migrations/DependencyFactory.php

```diff
@@ -197,9 +215,13 @@ public function getMigrationRepository() : MigrationRepository
         });
     }
 
-    public function setMetadataStorageConfiguration(MetadataStorageConfiguration $metadataStorageConfigration) : void
+    /**
+     * @param mixed $service
```

Comment:
I think you can't have anything but an object here, right?

```suggestion
     * @param object $service
```

Reply:
I think `callable` is a possible value for the migrations sorter... of course can be wrapped by an object via `__invoke`....

Reply:
If that's the case, the phpdoc at the beginning of the file is wrong, so please use the same thing in both places (either `mixed` or `object|callable` or `object`)

Reply:
You raised a good point. Actually there is a `setSorter` method responsible for setting a custom sorter. Does it make sense to have `setSorter`? we could use `setService` and drop `setSorter` ... but we do not have an interface for the sorter, since is just any callback...

Ideas?

Reply:
I'd either leave `sorter` alone and use `object`, or add an interface and provide a `FromCallableSorter` implementation

Reply:
What about this approach https://github.com/doctrine/migrations/pull/876/commits/a89d28916fe0ce495438bb59a890611239a9445d ?

Reply:
Looks good 👍

Reply:
good!